endpoints.php

```diff
@@ -1630,6 +1638,9 @@ function uploadVideosToKaltura($request)
 
         update_option(KalturaUpload::IN_PROGRESS_OPTION, KalturaUpload::YES_VALUE);
         update_option(KalturaUpload::STOP_UPLOAD_OPTION, KalturaUpload::NO_VALUE, false);
+        update_option(KalturaUpload::UPLOAD_ERROR_OPTION, '');
+
+        add_action('shutdown', 'cleanUpKalturaUpload');
```

Comment:
Does this add the callback to the hook every time this uploadVideosToKaltura function is run? If so, it seems like initiating multiple uploads may result in several callbacks being registered. Although, I haven't confirmed if this is actually true.

Reply:
After reading about the PHP [register_shutdown_function](https://www.php.net/manual/en/function.register-shutdown-function.php), it seems like the registered callback is only for the current process? If so, then I think this works fine.

Reply:
Yes, I think it is only for the current process. I tested it with some print statements and the shutdown hook was only called when the upload endpoint was called.